### DISCUSSION TITLE
Regarding contact and friction behavior for flat foot humanoid in MuJoCo

### DISCUSSION THREAD

**Discussion by r3tro-t3ch** [google-deepmind/mujoco#2175] [Asking for Help]
### Intro

I am Vishnu Joshi a researcher in StochLab, Indian Institute of Science, Bengaluru. Me and my team are currently working on a walking controller using MuJoCo simulator and had a few observations. 

### My setup

MuJoCo version 3.1.4, python, ubuntu 22.04

### My question

When the floor is a simple box geom with friction coefficient (0.8), the robot walks properly and has no issue, but when the same floor is defined by a hfield with the same friction coefficient (0.8) the foot is unable to stay stable.
I realised that increasing the nrow and ncol variable did improve the situation but it is not similiar to the behaviour when the floor is simple box geom. I tried going through all the resources and found that the number of contacts between geom and hfield is restricted to 50 and changing it can break the simulation.

I have attached two videos, one with simple box geom and other with hfield with respective names. 

https://github.com/user-attachments/assets/94b4eb95-dad3-4c83-8f03-38440576d73a

https://github.com/user-attachments/assets/263bd6ae-135e-477b-bb8e-0f8d67f22883

### Confirmations

- [X] I searched the [latest documentation](https://mujoco.readthedocs.io/en/latest/overview.html) thoroughly before posting.
- [X] I searched previous [Issues](https://github.com/google-deepmind/mujoco/issues) and [Discussions](https://github.com/google-deepmind/mujoco/discussions), I am certain this has not been raised before.

**Comment by yuvaltassa** [comment]
Read these links:
https://mujoco.readthedocs.io/en/latest/XMLreference.html#option-flag-multiccd
https://mujoco.readthedocs.io/en/latest/modeling.html#preventing-slip
